Src/DependencyCollector/Shared/Implementation/ClientServerDependencyTracker.cs

```diff
@@ -32,6 +32,18 @@ internal static DependencyTelemetry BeginTracking(TelemetryClient telemetryClien
             // but every operation must have it's own Activity and Id must be set accordingly
             // basically we repeat TelemetryClientExtensions.StartOperation here
             var activity = new Activity(DependencyActivityName);
+
+            // This is workaround for the issue https://github.com/Microsoft/ApplicationInsights-dotnet/issues/538
+            // if there is no parent Activity, ID Activity generates is not random enough to work well with 
+            // ApplicationInsights sampling algorithm
+            // This code should go away when Activity is fixed: https://github.com/dotnet/corefx/issues/18418
+            if (Activity.Current == null)
```

Comment:
StartOperation in fact uses ([here](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/151509eac5e23fcbd09f3a8989a35d1099703957/src/Core/Managed/Shared/TelemetryClientExtensions.cs#L97) and later [here](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/151509eac5e23fcbd09f3a8989a35d1099703957/src/Core/Managed/Shared/TelemetryClientExtensions.cs#L129)) OperationTelemetry.Id if there was no parent Activity. So, StartOperation is free from the sampling problems.

If the root, top-most Activity was created with StartOperation, everythin is fine, if it was created in  Web SDK or AspNetCore SDK, we make sure parent activity is fine in the SDKs.

If user created Activity manually, sampling may not work well, but it is the same as if user passed custom operationId in the StartOperation or set it manually in the telemetry.